router/src/lib.rs

```diff
@@ -906,8 +906,18 @@ impl ChatRequest {
         let (inputs, grammar, using_tools) = prepare_chat_input(
             infer,
             response_format,
-            tools,
-            tool_choice,
+            tools.clone(),
```

Comment:
No clone.

Reply:
removed, it was previously required because of a check of the `tools` vec that was below this line (moved it up and removed the clone). thanks